backend/src/main/scala/cromwell/backend/validation/ContinueOnReturnCodeValidation.scala

```diff
@@ -0,0 +1,46 @@
+package cromwell.backend.validation
+
+import cromwell.backend.validation.RuntimeAttributesValidation._
+import cromwell.core._
+import wdl4s.types.{WdlArrayType, WdlStringType}
+import wdl4s.values.{WdlArray, WdlBoolean, WdlInteger, WdlString}
+
+import scalaz.Scalaz._
+
+trait ContinueOnReturnCodeValidation extends RuntimeAttributesValidation[ContinueOnReturnCode] {
+  override def key = RuntimeAttributesKeys.ContinueOnReturnCodeKey
+
+  override def staticDefaultOption = Option(WdlInteger(0))
+
+  override def coercion = ContinueOnReturnCode.validWdlTypes
+
+  override protected def failureMessage = {
+    s"Expecting $key runtime attribute to be either a Boolean, a String 'true' or 'false', or an Array[Int]"
+  }
+
+  override def validateValue = {
+    case WdlBoolean(value) => ContinueOnReturnCodeFlag(value).successNel
+    case WdlString(value) if value.toLowerCase == "true" => ContinueOnReturnCodeFlag(true).successNel
+    case WdlString(value) if value.toLowerCase == "false" => ContinueOnReturnCodeFlag(false).successNel
+    case WdlInteger(value) => ContinueOnReturnCodeSet(Set(value)).successNel
+    case WdlArray(wdlType, seq) =>
+      val nels: Seq[ErrorOr[Int]] = seq map validateInt
+      val numFailures: Int = nels.count(_.isFailure)
+      if (numFailures == 0) {
+        val defaultReturnCodeNel = Set.empty[Int].successNel[String]
+        nels.foldLeft(defaultReturnCodeNel)((acc, v) => (acc |@| v) { (a, v) => a + v }) map ContinueOnReturnCodeSet
+      } else {
+        failureWithMessage
+      }
```

Comment:
I first learned of `sequence` through the same Coursera class you mention in the context of transforming `List[Future[A]]` to `Future[List[A]]`.  And as you also mention Slick offers a similar facility for `Seq[DBIO[A]]` to `DBIO[Seq[A]]`.

In picking up Haskell I learned the concept of transforming a `F[G[A]]` to `G[F[A]]` is captured by the `Traversable` typeclass.  This is called `Traverse` in both Scalaz and Cats as the `Traversable` name is already taken by a trait in the standard library.

Scalaz appears to put its "batteries included" typeclasses for standard library classes in scalaz.std, and there's a `ListInstances` there in `List.scala` that provides a `Traverse` instance for `List`.  The signature of `sequence` looks like:

```scala
  /** Traverse with the identity function. */
  def sequence[G[_]:Applicative,A](fga: F[G[A]]): G[F[A]] =
    traversal[G].run[G[A], A](fga)(ga => ga)
```

The role of `F` here is played by `List`, so as long as `G` has an `Applicative` instance it will be possible to transform `F[G[A]]` to `G[F[A]]`.  The role of `G` here is played by `NonEmptyList`, which in its source file has a `Monad` instance, and `Monad` extends `Applicative`.

I haven't found a really effective java to scalaz translation either.  Instead I've found it a lot easier to learn concepts in Haskell and then figure out how they might be applied to Scala using Scalaz or Cats.